cg/services/illumina/backup/utils.py

```diff
@@ -0,0 +1,118 @@
+"""Helper functions for the back-up of Illumina flow cells."""
+
+from datetime import datetime
+from operator import attrgetter
+from pathlib import Path
+
+from cg.constants import FileExtensions
+from cg.meta.backup.backup import LOG
+from cg.services.illumina.backup.models import PdcEncryptionKey, PdcSequencingFile
+from cg.utils.date import convert_string_to_datetime_object
+
+
+class DsmcOutput:
+    DATE_COLUMN_INDEX = 2
+    TIME_COLUMN_INDEX = 3
+    PATH_COLUMN_INDEX = 4
+
+
+def get_latest_archived_sequencing_run_path(dsmc_output: list[str]) -> Path | None:
+    """Get the path of the archived sequencing run from a PDC query."""
+    validated_sequencing_paths: list[PdcSequencingFile] = _parse_dsmc_output_sequencing_path(
+        dsmc_output
+    )
+    archived_run: PdcSequencingFile | None = _get_latest_sequencing_run_in_list(
+        validated_sequencing_paths
+    )
+    if archived_run:
+        LOG.info(f"Sequencing run found: {archived_run}")
+        return archived_run.path
+
+
+def get_latest_archived_encryption_key_path(dsmc_output: list[str]) -> Path | None:
+    """Get the encryption key for the archived sequencing run from a PDC query."""
+    validated_encryption_keys: list[PdcEncryptionKey] = _parse_dsmc_output_key_path(dsmc_output)
+
+    archived_encryption_key: PdcEncryptionKey | None = _get_latest_pdc_encryption_key(
+        validated_encryption_keys
+    )
+    if archived_encryption_key:
+        LOG.info(f"Encryption key found: {archived_encryption_key}")
+        return archived_encryption_key.path
+
+
+def _parse_dsmc_output_sequencing_path(dsmc_output: list[str]) -> list[PdcSequencingFile]:
+    """Parses the DSMC command output to extract validated sequencing paths."""
+    validated_responses = []
+    for line in dsmc_output:
+        if is_pdc_sequencing_path(line):
+            parts = line.split()
+
+            file_date_time: datetime = convert_string_to_datetime_object(
+                f"{parts[DsmcOutput.DATE_COLUMN_INDEX]} {parts[DsmcOutput.TIME_COLUMN_INDEX]}"
+            )
+
+            query_response: PdcSequencingFile = PdcSequencingFile(
```

Comment:
we omit the type hint if it is followed by the class constructor
```suggestion
            query_response: PdcSequencingFile(
```